lib/mk-ca-bundle.pl

```diff
@@ -436,9 +436,25 @@ (%)
       last if (/\*\*\*\*\* END LICENSE BLOCK \*\*\*\*\*/);
     }
   }
-  elsif(/^# (Issuer|Serial Number|Subject|Not Valid Before|Not Valid After |Fingerprint \(MD5\)|Fingerprint \(SHA1\)):/) {
+# Not Valid After : Thu Sep 30 14:01:15 2021
+  elsif(/^# Not Valid After : (.*)/) {
```

Comment:
why is this no longer pushed to precert, also if there is no `Not Valid After` in the header (which shouldn't happen afaict) then the last valid state is reused?

Reply:
> why is this no longer pushed to precert

That's a mistake I think. I never use `-m` myself and we have no tests for this... 

> also if there is no Not Valid After in the header

I thought about it but I could come up with a quick fix so I left it like this since I wanted to land this and run the script asap. Any suggestions on how to improve this? The script is a little quirky...

Reply:
The logic a few lines down of resetting precert doesn't make sense to me:
~~~perl
  elsif(/^#|^\s*$/) {
      undef @precert;
      next;
  }
~~~
Which is to say that if # is detected and does not match any known header (because otherwise this code wouldn't be reached) that it resets precert, or if there is a blank line it resets precert. For the former case wouldn't we want to keep precert even if some unknown header is found?

To your question I think that valid should start off as 0 and reset to 0 at the beginning of processing every certificate but I'm having trouble finding exactly where it makes that determination. I thought it could be in the code quoted above but a `# Certificate ` check would make more sense imo

Reply:
Feel free to work on improvements!